src/Combinatorics/Matroids/quantum_automorphism_groups.jl

```diff
@@ -0,0 +1,180 @@
+  sizes = unique!(map(length, sets))
+  tempGrdSet = reduce(vcat,[grdSet for i in 1:n])
+
+  for size in sizes 
+    size == 0 && continue
+    powerSet = unique(sort.(subsets(tempGrdSet, size)))
+
+    setsOfSize = filter(x->length(x)==size, sets)
+    nonSets = setdiff(powerSet, setsOfSize) 
+
+    S = symmetric_group(size)
+
+    for set in setsOfSize
+      append!(b[size], [[set[g(i)] for i in 1:size] for g in S])
+    end
+    for nonset in nonSets
+      append!(nb[size], [[nonset[g(i)] for i in 1:size] for g in S])
+    end
+    for set in b[size]
+      for nonset in nb[size]
+        push!(rels[size], [(set[i],nonset[i]) for i in 1:size])
+        push!(rels[size], [(nonset[i],set[i]) for i in 1:size])
+      end
+    end
+  end
+  rels = unique!.(rels)
+  return Vector{Vector{Tuple{Int,Int}}}(reduce(vcat,rels))
+end
+
+@doc raw"""
+    quantum_automorphism_group(M::Matroid, structure::Symbol=:bases)
+
+Return the Ideal that defines the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+
+```jldoctest
+julia> G = complete_graph(4)
+Undirected graph with 4 nodes and the following edges:
+(2, 1)(3, 1)(3, 2)(4, 1)(4, 2)(4, 3)
+
+julia> M = cycle_matroid(G);
+
+julia> qAut = quantum_automorphism_group(M,:bases);
+
+julia> length(gens(qAut))
+23448
+```
+"""
+function quantum_automorphism_group(
+  M::Matroid,
+  structure::Symbol=:bases)
+
+  relation_indices = _quantum_automorphism_group_indices(M, structure)
+  n = length(M)
+
+  SN = quantum_symmetric_group(n)
+  A = base_ring(SN)
+  u = permutedims(reshape(gens(A),(n, n)),[2, 1])
+
+  new_relations = elem_type(A)[prod(gen -> u[gen[1], gen[2]], relation; init=one(A)) for relation in relation_indices]
+  return ideal([gens(SN)..., new_relations...])
+
+end
+
+@doc raw"""
+    quantum_automorphism_group(G::Graph)
+
+Return the Ideal that defines the quantum automorphism group of a graph.
+
+# Examples
+```jldoctest
+julia> G = graph_from_edges([[1, 2], [2, 4]]);
+
+julia> qAut = quantum_automorphism_group(G);
+
+julia> length(gens(qAut))
+216
+```
+"""
+function quantum_automorphism_group(G::Graph{Undirected})
+
+  n = nv(G)
+
+  SN = quantum_symmetric_group(n)
+  A = base_ring(SN)
+  u = permutedims(reshape(gens(A),(n, n)),[2, 1])
+
+  nonedges = Tuple{Int,Int}[(i, j) for i in 1:n for j in i:n if !has_edge(G,i,j)] # This should be n^2 - |E| many nonedges
+
+  edgs = map(edg -> (src(edg), dst(edg)), edges(G)) # This should be |E| many edges
+
+  function _addrelations(edge,nonedg)
+    r1 = u[edge[1], nonedg[1]] * u[edge[2], nonedg[2]]
+    r2 = u[edge[2], nonedg[1]] * u[edge[1], nonedg[2]]
+    r3 = u[edge[1], nonedg[2]] * u[edge[2], nonedg[1]] 
+    r4 = u[edge[2], nonedg[2]] * u[edge[1], nonedg[1]]
+    return [r1,r2,r3,r4]
+  end
+  new_relations = reduce(vcat,[vcat(_addrelations(edg, nonedg), _addrelations(nonedg, edg)) for edg in edgs for nonedg in nonedges])
+  return ideal([gens(SN)..., unique(new_relations)...])
```

Comment:
To be clear, the following is relatively minor. But: if the goal is to make this code optional, then it shouldn't use `vcat` and `reduce, but rather a straight-forward loop pushing things into a vector for which the max capacity was set beforehand. Like this change perhaps (untested), which avoids allocation a length 4 vector for each call of `_addrelations`, followed by another allocation for the inner `vcat`, etc. etc. -- this function should only allocate for the result vector perhaps two times.
```suggestion
  new_relations = copy(gens(SN))
  sizehint!(new_relations, ngens(SN) + 8*length(edgs)*length(nonedges))
  function _addrelations(edge, nonedg)
    r1 = u[edge[1], nonedg[1]] * u[edge[2], nonedg[2]]
    r2 = u[edge[2], nonedg[1]] * u[edge[1], nonedg[2]]
    r3 = u[edge[1], nonedg[2]] * u[edge[2], nonedg[1]] 
    r4 = u[edge[2], nonedg[2]] * u[edge[1], nonedg[1]]
    push!(new_relations, r1, r2, r3, r4)
  end
  for edg in edgs, nonedg in nonedges
    _addrelations(edg, nonedg)
    _addrelations(nonedg, edg)
  end
  unique!(new_relations)
  return ideal(new_relations)
```

Reply:
I applied this suggestion.